File: template_js/problema.test.js
// Mantenha o seu cursor aqui: |

const mapping = [
    [1000, 'M'],
    [500, 'D'],
    [400, 'CD'],
    [100, 'C'],
    [90, 'XC'],
    [50, 'L'],
    [40, 'XL'],
    [10, 'X'],
    [9, 'IX'],
    [5, 'V'],
    [4, 'IV'],
    [1, 'I'],
]

function romano(number) {
    let ret = ''

    mapping.forEach(([arabic, roman]) => {
        while (number >= arabic) {
            ret += roman
            number -= arabic
        }
    })

    return ret
}

test('1485 == MCDLXXXV', () => {
    expect(romano(1485)).toBe('MCDLXXXV');
});

test('998 == CM', () => {
    expect(romano(499)).toBe('CDXCIX');
});

test('499 == CDXCIX', () => {
    expect(romano(499)).toBe('CDXCIX');
});

test('99 == XCIX', () => {
    expect(romano(99)).toBe('XCIX');
});

test('49 == XLIX', () => {
    expect(romano(49)).toBe('XLIX');
});

test('9 == IX', () => {
    expect(romano(9)).toBe('IX');
});

test('358 == CCCLVIII', () => {
    expect(romano(358)).toBe('CCCLVIII');
});

test('4 == IV', () => {
    expect(romano(4)).toBe('IV');
});

test('51 == LI', () => {
    expect(romano(51)).toBe('LI');
});

test('1 == I', () => {
    expect(romano(1)).toBe('I');
});

test('2 == II', () => {
    expect(romano(2)).toBe('II');
});

test('3 == III', () => {
    expect(romano(3)).toBe('III');
});

test('5 == V', () => {
    expect(romano(5)).toBe('V');
});

test('10 == X', () => {
    expect(romano(10)).toBe('X');
});

test('20 == XX', () => {
    expect(romano(20)).toBe('XX');
});

test('30 == XXX', () => {
    expect(romano(30)).toBe('XXX');
});

test('500 == D', () => {
    expect(romano(500)).toBe('D');
});

test('100 == C', () => {
    expect(romano(100)).toBe('C');
});

test('200 == CC', () => {
    expect(romano(200)).toBe('CC');
});

test('300 == CCC', () => {
    expect(romano(300)).toBe('CCC');
});

test('1000 == M', () => {
    expect(romano(1000)).toBe('M');
});

test('2000 == MM', () => {
    expect(romano(2000)).toBe('MM');
});

test('3000 == MMM', () => {
    expect(romano(3000)).toBe('MMM');
});

File: templategolang/problema_test.go
package templategolang

import "testing"

func TestMain(t *testing.T) {
	tests := []struct {
		name     string
		number   Arabic
		expected Roman
	}{
		{
			name:     "1485 == MCDLXXXV",
			number:   1485,
			expected: "MCDLXXXV",
		},
		{
			name:     "998 == CM",
			number:   499,
			expected: "CDXCIX",
		},
		{
			name:     "499 == CDXCIX",
			number:   499,
			expected: "CDXCIX",
		},
		{
			name:     "99 == XCIX",
			number:   99,
			expected: "XCIX",
		},
		{
			name:     "49 == XLIX",
			number:   49,
			expected: "XLIX",
		},
		{
			name:     "9 == IX",
			number:   9,
			expected: "IX",
		},
		{
			name:     "358 == CCCLVIII",
			number:   358,
			expected: "CCCLVIII",
		},
		{
			name:     "4 == IV",
			number:   4,
			expected: "IV",
		},
		{
			name:     "51 == LI",
			number:   51,
			expected: "LI",
		},
		{
			name:     "1 == I",
			number:   1,
			expected: "I",
		},
		{
			name:     "2 == II",
			number:   2,
			expected: "II",
		},
		{
			name:     "3 == III",
			number:   3,
			expected: "III",
		},
		{
			name:     "5 == V",
			number:   5,
			expected: "V",
		},
		{
			name:     "10 == X",
			number:   10,
			expected: "X",
		},
		{
			name:     "20 == XX",
			number:   20,
			expected: "XX",
		},
		{
			name:     "30 == XXX",
			number:   30,
			expected: "XXX",
		},
		{
			name:     "500 == D",
			number:   500,
			expected: "D",
		},
		{
			name:     "100 == C",
			number:   100,
			expected: "C",
		},
		{
			name:     "200 == CC",
			number:   200,
			expected: "CC",
		},
		{
			name:     "300 == CCC",
			number:   300,
			expected: "CCC",
		},
		{
			name:     "1000 == M",
			number:   1000,
			expected: "M",
		},
		{
			name:     "2000 == MM",
			number:   2000,
			expected: "MM",
		},
		{
			name:     "3000 == MMM",
			number:   3000,
			expected: "MMM",
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := romano(tt.number); got != tt.expected {
				t.Errorf("main() = %v, want %v", got, tt.expected)
			}
		})
	}
}
